includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -73,23 +73,40 @@ class AMP_Gallery_Block_Sanitizer extends AMP_Base_Sanitizer {
 	 * @since 0.2
 	 */
 	public function sanitize() {
-		$nodes     = $this->dom->getElementsByTagName( self::$tag );
-		$num_nodes = $nodes->length;
-		if ( 0 === $num_nodes ) {
-			return;
-		}
+		$xpath = new DOMXPath( $this->dom );
+		$expr  = sprintf(
+			'//ul[ %s ]',
+			implode(
+				' or ',
+				[
+					'( parent::figure[ contains( @class, "wp-block-gallery" ) and @data-amp-carousel = "true" ] )',
+					'( contains( @class, "wp-block-gallery" ) and @data-amp-carousel = "true" )',
+				]
+			)
+		);
+		$query = $xpath->query( $expr );
 
-		for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
-			$node = $nodes->item( $i );
+		$nodes = [];
+		foreach ( $query as $node ) {
+			$nodes[] = $node;
+		}
 
-			// We're looking for <ul> elements that have at least one child and the proper class.
-			if ( 0 === count( $node->childNodes ) || false === strpos( $node->getAttribute( 'class' ), self::$class ) ) {
-				continue;
-			}
+		foreach ( $nodes as $node ) {
+			/**
+			 * Element
+			 *
+			 * @var DOMElement $node
+			 */
 
 			$attributes      = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $node );
 			$is_amp_lightbox = isset( $attributes['data-amp-lightbox'] ) && true === filter_var( $attributes['data-amp-lightbox'], FILTER_VALIDATE_BOOLEAN );
```

Comment:
Like `$is_amp_carousel`, `$is_amp_lightbox` also needs to check the `parentNode` to see if `data-amp-lightbox` is set (for Gutenberg 6.5+ compatibility). The simplest way of handling this situation for both `$is_amp_carousel` and `$is_amp_lightbox` might be something like this:
```
$attributes      = array_merge( AMP_DOM_Utils::get_node_attributes_as_assoc_array( $node ), AMP_DOM_Utils::get_node_attributes_as_assoc_array( $node->parentNode ) );
```

Reply:
Hi @claudiulodro, thanks for pointing that out. I ended up doing something similar to that.